internal/api/handler/updatemetrics_test.go

```diff
@@ -0,0 +1,373 @@
+)
+
+func TestMetricsHandler_UpdateMetric(t *testing.T) {
+	metricUseCase := mocks.NewUseCase(t)
+	h := NewMetricsHandler(metricUseCase)
+
+	type (
+		input struct {
+			method      string
+			query       string
+			metricType  string
+			metricName  string
+			metricValue string
+		}
+		want struct {
+			statusCode int
+			response   string
+		}
+	)
+
+	tests := []struct {
+		name  string
+		input input
+		want  want
+	}{
+		{
+			name: "non POST method",
+			input: input{
+				method: http.MethodGet,
+				query:  "/update/gauge/metric1/value",
+			},
+			want: want{
+				statusCode: http.StatusMethodNotAllowed,
+			},
+		},
+		{
+			name: "missing parameters",
+			input: input{
+				method: http.MethodPost,
+				query:  "/update/",
+			},
+			want: want{
+				statusCode: http.StatusNotFound,
+			},
+		},
+		{
+			name: "undefined metric type",
+			input: input{
+				method:      http.MethodPost,
+				query:       "/update/undefined_type/metric1/123.45",
+				metricType:  "undefined_type",
+				metricName:  "metric1",
+				metricValue: "123.45",
+			},
+			want: want{
+				statusCode: http.StatusBadRequest,
+			},
+		},
+		{
+			name: "successful metric update",
+			input: input{
+				method:      http.MethodPost,
+				query:       "/update/gauge/metric1/123.45",
+				metricType:  "gauge",
+				metricName:  "metric1",
+				metricValue: "123.45",
+			},
+			want: want{
+				statusCode: http.StatusOK,
+				response:   `{"message":"Metric updated successfully"}`,
+			},
+		},
+	}
+
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			metricUseCase.ExpectedCalls = nil
+			metricUseCase.Calls = nil
+
+			r := httptest.NewRequest(tt.input.method, tt.input.query, nil)
+			ctx := chi.NewRouteContext()
+			ctx.URLParams.Add("type", tt.input.metricType)
+			ctx.URLParams.Add("name", tt.input.metricName)
+			ctx.URLParams.Add("value", tt.input.metricValue)
+			r = r.WithContext(context.WithValue(r.Context(), chi.RouteCtxKey, ctx))
+
+			w := httptest.NewRecorder()
+
+			switch tt.name {
+			case "undefined metric type":
+				metricUseCase.On("Update", r.Context(), entity.MetricDTO{
+					Type:  "undefined_type",
+					Name:  "metric1",
+					Value: "123.45",
+				}).Return(metricusecase.UpdateResp{}, errorz.ErrUndefinedMetricType)
+			case "undefined metric name":
+				metricUseCase.On("Update", r.Context(), entity.MetricDTO{
+					Type:  "gauge",
+					Name:  "undefined_name",
+					Value: "123.45",
+				}).Return(metricusecase.UpdateResp{}, errorz.ErrUndefinedMetricName)
+			case "successful metric update":
+				metricUseCase.On("Update", r.Context(), entity.MetricDTO{
+					Type:  "gauge",
+					Name:  "metric1",
+					Value: "123.45",
+				}).Return(metricusecase.UpdateResp{
+					Message: "Metric updated successfully",
+				}, nil)
+			}
+
+			h.UpdateMetric(w, r)
+			res := w.Result()
+			assert.Equal(t, tt.want.statusCode, res.StatusCode)
+			if tt.want.response != "" {
+				body, _ := io.ReadAll(res.Body)
```

Comment:
Не стоит игнорировать обработку ошибки. Можно  
```suggestion
				body, err := io.ReadAll(res.Body)
				if err != nil {
					t.Errorf("error reading response body: %v", err)
				}
```